### PR TITLE
improved optional value error message

### PR DESCRIPTION
**Pull request by sanderr**

### CODE REVIEW

**Comment by sanderr** [comment]
Old error message:
```
inmanta.ast.OptionalValueException: Optional variable accessed that has no value (aws::VirtualMachine (instantiated at {dir}/main.cf:9).aws::VirtualMachine.subnet) (reported in self.subnet ({dir}/main.cf:9))
```
New error message:
```
inmanta.ast.OptionalValueException: Optional variable accessed that has no value (attribute `aws::VirtualMachine.subnet` of `aws::VirtualMachine (instantiated at {dir}/main.cf:6)`) (reported in self.subnet ({dir}/libs/aws/model/_init.cf:105))
```

File: src/inmanta/execute/runtime.py [hunk 1]
```diff
@@ -710,6 +710,7 @@ def execute(self) -> None:
             self.resumer.resume(self.requires, self.resolver, self.queue)
         except RuntimeException as e:
             e.set_statement(self.resumer)
+            e.location = self.resumer.location
```
Comment (wouterdb) [inline src/inmanta/execute/runtime.py]: shouldn't set_statement set the location?

Comment (sanderr) [reply src/inmanta/execute/runtime.py]: I'm not sure. It doesn't when it's already been set at least. Is this a bug?

Comment (wouterdb) [reply src/inmanta/execute/runtime.py]: no, this is correct. So this is a hard override.

**Pull request merged**